gnovm/pkg/gnolang/preprocess.go

```diff
@@ -430,10 +430,15 @@ func Preprocess(store Store, ctx BlockNode, n Node) Node {
 				checkValDefineMismatch(n)
 
 				if n.Op == DEFINE {
-					for _, lx := range n.Lhs {
+					for i, lx := range n.Lhs {
```

Comment:
Why is it necessary to conduct checks on different sites at various stages? can you provide more tests to support this?

Reply:
After checking again, it wasn't particulary relevant. so removed it.
https://github.com/gnolang/gno/pull/2699/commits/9b8f6971131523a4af78ca647ad8c3d7ee5c1319